lib/Server.js

```diff
@@ -62,13 +62,54 @@ class Server {
     return /^[a-zA-Z][a-zA-Z\d+\-.]*:/.test(URL);
   }
 
+  static findIp(gateway) {
+    const gatewayIp = ipaddr.parse(gateway);
+
+    // Look for the matching interface in all local interfaces.
+    for (const addresses of Object.values(os.networkInterfaces())) {
+      for (const { cidr } of addresses) {
+        const net = ipaddr.parseCIDR(cidr);
+
+        if (
+          net[0] &&
+          net[0].kind() === gatewayIp.kind() &&
+          gatewayIp.match(net)
+        ) {
+          return net[0].toString();
+        }
+      }
+    }
+  }
+
+  static async asyncInternalIp(family) {
+    try {
+      const { gateway } = await defaultGateway[family]();
+      return Server.findIp(gateway);
+    } catch {
+      // ignore
+    }
+  }
+
+  static syncInternalIp(family) {
+    try {
+      const { gateway } = defaultGateway[family].sync();
+      return Server.findIp(gateway);
+    } catch {
+      // ignore
+    }
+  }
```

Comment:
Let's use normal name for `async` (i.e. `internalIP`)  and `*Sync` for sync (i.e. `internalIPSync`)